bgpd/bgp_flowspec.c

```diff
@@ -150,12 +149,8 @@ int bgp_nlri_parse_flowspec(struct peer *peer, struct attr *attr,
 		p.family = AF_FLOWSPEC;
 		p.prefixlen = 0;
 		/* Flowspec encoding is in bytes */
-		p.u.prefix_flowspec.prefixlen = psize;
 		p.u.prefix_flowspec.family = afi2family(afi);
-		temp = XCALLOC(MTYPE_TMP, psize);
-		memcpy(temp, pnt, psize);
-		p.u.prefix_flowspec.ptr = (uintptr_t) temp;
-
+		prefix_flowspec_allocate(&p, pnt, psize);
```

Comment:
so this is ... malloc'ing some memory and attaching it to a prefix struct that's on the stack? how could you call the corresponding 'free' function in this case?

Reply:
right now, it is up to the user to pay attention.

Reply:
I ... don't understand that reply? 'p' is on the stack; the prefix_free() api is never called - right?

Reply:
furthermore, the copy that's made in table.c:route_node_get() isn't freed either - how could it be, again it's on the stack.

Reply:
I'm not certain this allocated info belongs in the prefix struct at all - how about putting it into the bgp_dest/bgp_node struct: that has a real dtor?

Reply:
actually, sometimes, the user does not call prefix_free(), and directly call XFREE(..)
reworking all the code is not the goal of this ticket.

your suggestion to move extra data  out of the prefix, I want to chat with you to understand more how that can be used ?.